docker-compose.yml

```diff
@@ -0,0 +1,26 @@
+version: '3.8'
+services:
+  app:
+    build:
+      context: .
+      args:
+        NODE_ENV: development
+    volumes:
+      - node_modules:/app/node_modules
+      - dist:/app/dist
+      - .:/app:delegated
+    command: npm run dev
+    ports:
+      - 3000:3000
+
+volumes:
+  node_modules:
```

Comment:
This is where the magic happens. specifying a named bind volume pointing the "device" to the repo equivalent folder. This Allows the volume to be accessible on the host at the given path.